### PR TITLE
chore(wallet): avoid chain/account switches in Send

### PR DESCRIPTION
**Pull request by onyb**
This PR does the following two refactorings:

1. Send screen no longer relies on `selectedNetwork` and `selectedAccount` backend states.
2. Field information in Send is encoded in the URL query params instead of relying on `React.useState`. This allowed us to cleanup the`useEffect` mess in Send screen.

Resolves https://github.com/brave/brave-browser/issues/34375

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Send flow in desktop and android should continue to work.
2. Must be tested for Ethereum, Solana, Filecoin.
3. Must be tested for ERC20s/SPL and NFTs.
4. Verify whether selecting an asset changes the query params in the URL.
5. Go to **Portfolio** and group entries by **Accounts**. Clicking on Send for an asset should populate it in the Send screen.
6. Open an account under the **Accounts** page. Clicking on Send for an asset should populate it in the Send screen.

### CODE REVIEW

File: components/brave_wallet_ui/utils/account-utils.ts [hunk 1]
```diff
@@ -99,7 +99,23 @@ export const findAccountByAccountId = (
   if (!accounts) {
     return undefined
   }
-  return accounts.entities[entityIdFromAccountId(accountId)]
+
```
Comment (onyb) [inline components/brave_wallet_ui/utils/account-utils.ts:102]: @josheleonard Need your eyes on this block.

Comment (josheleonard) [reply components/brave_wallet_ui/utils/account-utils.ts:102]: LGTM

File: components/brave_wallet_ui/page/screens/send/send_screen/send_screen.tsx [hunk 2]
```diff
@@ -146,28 +145,31 @@ export const SendScreen = React.memo((props: Props) => {
...
+  const { data: networks = [] } = useGetVisibleNetworksQuery()
+  const networkFromParams = React.useMemo(
+    () =>
+      networks.find(
+        (network) =>
+          network.chainId === query.get('chainId') &&
+          network.coin === accountFromParams?.accountId.coin
+      ),
+    [networks, accountFromParams, query]
+  )
```
Comment (josheleonard) [inline components/brave_wallet_ui/page/screens/send/send_screen/send_screen.tsx:166]: nit: can we group these with the others under `// queries` ?

File: components/brave_wallet_ui/page/screens/send/send_screen/send_screen.tsx [hunk 3]
```diff
@@ -178,84 +180,83 @@ export const SendScreen = React.memo((props: Props) => {
...
           token.tokenId === tokenId
-        : (token.contractAddress.toLowerCase() === contractOrSymbolLower &&
-            token.chainId === chainId) ||
-          (token.symbol.toLowerCase() === contractOrSymbolLower &&
-            token.chainId === chainId &&
+        : (token.contractAddress.toLowerCase() ===
+            contractOrSymbol.toLowerCase() &&
+            token.chainId === networkFromParams.chainId) ||
+          (token.symbol.toLowerCase() === contractOrSymbol.toLowerCase() &&
+            token.chainId === networkFromParams.chainId &&
```
Comment (josheleonard) [inline components/brave_wallet_ui/page/screens/send/send_screen/send_screen.tsx:219]: pref nit: probably faster to check chainID before the symbol/contract?

**Review by josheleonard — APPROVED** [review]
++ Good work!

**Pull request merged**